File: frontend/demo/component/notification/notification-position.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/button';
import { Notification } from '@vaadin/notification';
import type { NotificationPosition } from '@vaadin/notification';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('notification-position')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  constructor() {
    super();
    this.classList.add('notification-position-example');
  }

  // tag::snippet[]
  protected override render() {
    return html`
      <vaadin-button @click="${this.show}">top-stretch</vaadin-button>
      <vaadin-button @click="${this.show}">top-start</vaadin-button>
      <vaadin-button @click="${this.show}">top-center</vaadin-button>
      <vaadin-button @click="${this.show}">top-end</vaadin-button>
      <vaadin-button @click="${this.show}">middle</vaadin-button>
      <vaadin-button @click="${this.show}">bottom-start</vaadin-button>
      <vaadin-button @click="${this.show}">bottom-center</vaadin-button>
      <vaadin-button @click="${this.show}">bottom-end</vaadin-button>
      <vaadin-button @click="${this.show}">bottom-stretch</vaadin-button>
    `;
  }

  show(event: MouseEvent) {
    // Use the button label as the location
    const position = (event.target as HTMLElement).textContent as NotificationPosition;

    Notification.show(position, { position });
  }

  // end::snippet[]
}

File: src/main/java/com/vaadin/demo/component/notification/NotificationPosition.java
package com.vaadin.demo.component.notification;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.notification.Notification;
import com.vaadin.flow.component.notification.Notification.Position;
import com.vaadin.flow.router.Route;

import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("notification-position")
public class NotificationPosition extends Div {

  public NotificationPosition() {
    setClassName("notification-position-example");

    // tag::snippet[]
    add(
      createButton(Position.TOP_STRETCH),
      createButton(Position.TOP_START),
      createButton(Position.TOP_CENTER),
      createButton(Position.TOP_END),
      createButton(Position.MIDDLE),
      createButton(Position.BOTTOM_START),
      createButton(Position.BOTTOM_CENTER),
      createButton(Position.BOTTOM_END),
      createButton(Position.BOTTOM_STRETCH)
    );
    // end::snippet[]
  }

  // tag::createButton[]
  private Button createButton(Notification.Position position) {
    Button button = new Button(position.getClientName());
    button.addClickListener(event -> show(position));
    return button;
  }
  // end::createButton[]

  // tag::show[]
  private void show(Notification.Position position) {
    Notification.show(position.getClientName(), 5000, position);
  }
  // end::show[]

  public static class Exporter extends DemoExporter<NotificationPosition> { // hidden-source-line
  } // hidden-source-line
}
